message/nimessage.yml

```diff
@@ -0,0 +1,415 @@
+      summary: Subscribe to a topic
+      description: Subscribes to a topic to receive all messages broadcast with that topic name.
+      operationId: SubscribeToTopic
+      parameters:
+        - in: body
+          name: topic
+          description: The topic to subscribe to
+          required: true
+          schema:
+            $ref: '#/definitions/Topic'
+      responses:
+        200:
+          description: Success
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/sessions/{token}/subscriptions/unsubscribe:
+    parameters:
+      - $ref: '#/parameters/Token'
+    post:
+      tags:
+        - Topics
+      summary: Unsubscribe from a topic
+      description: Unsubscribes from a topic to stop receiving messages for that topic.
+      operationId: UnsubscribeFromTopic
+      parameters:
+        - in: body
+          name: topic
+          description: The topic to unsubscribe from
+          required: true
+          schema:
+            $ref: '#/definitions/Topic'
+      responses:
+        200:
+          description: Success
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+
+  /v1/sessions/{token}/messages:
+    parameters:
+      - $ref: '#/parameters/Token'
+    post:
+      tags:
+        - Message Queue
+      summary: Publish a message
+      description: Publishes a broadcast message to a specific topic. All of the topic's subscribers will receive a copy of the message.
+      operationId: PublishMessage
+      parameters:
+        - in: body
+          name: topicAndMessage
+          description: The message to publish, and the topic to publish to
+          required: true
+          schema:
+            $ref: '#/definitions/TopicAndMessage'
+      responses:
+        200:
+          description: Success
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+    get:
+      tags:
+        - Message Queue
+      summary: Read a message from the queue
+      description: Read a single message from the queue, if any.
+      operationId: ReadMessage
+      parameters:
+        - in: query
+          name: timeoutMilliseconds
+          description: The time to keep the connection open while waiting for a message
+          type: integer
+          required: false
+      responses:
+        200:
+          schema:
+            $ref: '#/definitions/TopicAndMessage'
+          description: Success
+        204:
+          description: No messages were available.
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+    delete:
+      tags:
+        - Message Queue
+      summary: Flush the message queue
+      description: Flushes the queue of received messages, discarding all messages without reading them.
+      operationId: DeleteMessages
+      responses:
+        200:
```

Comment:
This probably should have been a 204.  It would require a change on the webservice itself to fix it.

Reply:
I filed a CAR: http://nippm.natinst.com/itg/web/knta/crt/RequestDetail.jsp?REQUEST_ID=711012

Reply:
The same would apply to subscribe, unsubscribe, and publish.

Reply:
Updated the CAR